projects/route-analyzer/src/lib/route-analyzer.ts

```diff
@@ -301,4 +307,22 @@ function fixupLoadChildren(
         }
         return referencedModule.module;
     }
+
+    // Using absolute path prevents issues when two module files have the exact same name
+    function getLoadChildrenAbsolutePath(moduleFolderPath: string, loadChildrenImportPath: string): string {
+        let absolutePath: string;
+
+        // root filenames are user provided and may be relative (eg. see package json's gen-app-routes:tenant)
+        if (path.isAbsolute(moduleFolderPath)) {
+            absolutePath = path.join(moduleFolderPath, loadChildrenImportPath);
+        } else {
+            absolutePath = path.join(programDirectory, moduleFolderPath, loadChildrenImportPath);
+        }
+
+        // '.ts' is often omitted on TS imports.
```

Comment:
Isn't the ts extension disallowed in imports? The compiler is often looking for `.d.ts` files or it could even be a JS file? Can you explain in a comment when `.ts` is present?

Reply:
I got it mixed up with js. You are correct, this never happens... And never will happen. Found this read to be interesting: https://github.com/microsoft/TypeScript/issues/35589